spec/ParseQuery.FullTextSearch.spec.js

```diff
@@ -1,25 +1,14 @@
 'use strict';
 
-const MongoStorageAdapter = require('../lib/Adapters/Storage/Mongo/MongoStorageAdapter')
-  .default;
-const mongoURI =
-  'mongodb://localhost:27017/parseServerMongoAdapterTestDatabase';
-const PostgresStorageAdapter = require('../lib/Adapters/Storage/Postgres/PostgresStorageAdapter')
-  .default;
-const postgresURI =
-  'postgres://localhost:5432/parse_server_postgres_adapter_test_database';
+const Config = require('../lib/Config');
 const Parse = require('parse/node');
 const request = require('../lib/request');
 let databaseAdapter;
 
 const fullTextHelper = () => {
-  if (process.env.PARSE_SERVER_TEST_DB === 'postgres') {
-    if (!databaseAdapter) {
-      databaseAdapter = new PostgresStorageAdapter({ uri: postgresURI });
-    }
-  } else {
-    databaseAdapter = new MongoStorageAdapter({ uri: mongoURI });
-  }
+  const config = Config.get('test');
```

Comment:
I don't think this is necessary. Have you read the [Contributor Guide](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#run-your-tests-against-postgres-optional)

Does `PARSE_SERVER_TEST_DB=postgres npm test` work for you?

Reply:
@dplewis this test fails locally on my system using that command if I don’t have Postgres running on localhost using the standard port, 5432. This fix makes it work on any address/port Postgres is binded to by using the already defined PARSE_SERVER_TEST_DB environment variable. In the test cases current form, PARSE_SERVER_TEST_DB doesn’t work as documented without this fix. Two test fail in this case, which is I only fixed those.

You can replicate the failure by dowloading Postgres via docker and binding to a different port.

I suspect this is why the Travis file has to change the Postgres config to 5432 as the newly installed Postgres installations use 5433 and the PARSE_SERVER_TEST_DB doesn’t work properly.

Reply:
To my last point, after this pull request is reconciled, I plan on creating another one which edits the travis file and scripts to not have to stop/start postgres, nor have to remove older versions of postgres that are running on port 5432. Using PARSE_SERVER_TEST_DB should allow us to test on port 5433 directly after install of postgres without changing the image postgres config files (hopefully). I'm assuming this will speed up the postgres builds along with keeping the fix of the old postgres build issues.

If you think it's relevant enough, I can make the changes here if you think they are related enough and we can see how they go

Reply:
In addition, PARSE_SERVER_TEST_DATABASE_URI isn't mentioned in the [Contributors Guide](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#run-your-tests-against-postgres-optional), but it is in the [spec/helper.js](https://github.com/parse-community/parse-server/blob/da0dbfe23dfbb047fe84342da8c417f4ab882921/spec/ParseServer.spec.js) and [spec/ParseServer.js](https://github.com/parse-community/parse-server/blob/fd0b535159877fa9bc74fb645169f56814c67c60/spec/helper.js) files. I can add this in as well.

Reply:
Lastly, if anyone attempts to run a local test using the [official Postgres](https://hub.docker.com/_/postgres) (which PostGIS docker image uses), POSTGRES_PASSWORD has to be set or else the image won't build. This means that all local postgres tests using docker will require the use if PARSE_SERVER_TEST_DATABASE_URI as the default postgres URI set in the spec won't work. See my example below:

``` 
docker run -d --name parse-postgres -p 5432:5432 -e POSTGRES_USER=$USER -e POSTGRES_PASSWORD=password --rm postgis/postgis:11-3.0-alpine && sleep 20 && docker exec -it parse-postgres psql -U $USER -c 'CREATE DATABASE parse_server_postgres_adapter_test_database;' && docker exec -it parse-postgres psql -U $USER -c 'CREATE EXTENSION postgis;' -d parse_server_postgres_adapter_test_database && docker exec -it parse-postgres psql -U $USER -c 'CREATE EXTENSION postgis_topology;' -d parse_server_postgres_adapter_test_database
...
PARSE_SERVER_TEST_DB=postgres PARSE_SERVER_TEST_DATABASE_URI=postgres://$USER:password@localhost:5432/parse_server_postgres_adapter_test_database npm test``` 

Note that I didn't use the [million/postgis](https://hub.docker.com/r/mdillon/postgis/tags) image discussed in the Contributors Guide as it hasn't been updated in over a year.